mllib/src/main/scala/org/apache/spark/ml/regression/FMRegressor.scala

```diff
@@ -47,7 +47,7 @@ import org.apache.spark.storage.StorageLevel
  */
 private[ml] trait FactorizationMachinesParams extends PredictorParams
   with HasMaxIter with HasStepSize with HasTol with HasSolver with HasSeed
-  with HasFitIntercept with HasRegParam {
+  with HasFitIntercept with HasRegParam with HasWeightCol {
```

Comment:
Add ```with HasWeightCol``` because ```ClassificationSummary``` uses weigthCol. However, FM doesn't really support instance weight yet and all the weight are default to 1.0.